odoo/addons/base/models/ir_mail_server.py

```diff
@@ -264,7 +264,8 @@ def connect(self, host=None, port=None, user=None, password=None, encryption=Non
             smtp_port = tools.config.get('smtp_port', 25) if port is None else port
             smtp_user = user or tools.config.get('smtp_user')
             smtp_password = password or tools.config.get('smtp_password')
-            from_filter = tools.config.get('from_filter')
+            from_filter = self.env['ir.config_parameter'].sudo().get_param(
+                'mail.default.from.filter', tools.config.get('from_filter'))
```

Comment:
maybe more consistent?

```suggestion
                'mail.default.from_filter', tools.config.get('from_filter'))
```